massa-pool-worker/src/worker.rs

```diff
@@ -99,29 +144,78 @@ impl OperationPoolThread {
 
     /// Run the thread.
     fn run(self, config: PoolConfig) {
+        let buffer_swap_interval = config.operation_pool_swap_interval.to_duration();
+        let mut operation_pool_buffer = self.operation_pool.read().clone();
         let mut start_time = Instant::now();
         let tick = config.operation_pool_refresh_interval.to_duration();
+        let mut modified = false;
+        let mut last_buffer_swap = Instant::now();
         loop {
+            // refresh if needed
```

Comment:
we can use select! for stuff like that between ticker and recv channel. (see connectivity.rs for example)

Reply:
the advantage here is the priorization:
as soon as there are no messages to process OR if there are but we waited more than X time, we do the buffer swap